optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolver.java

```diff
@@ -149,16 +149,20 @@ public final void solve(Solution planningProblem) {
             throw new IllegalArgumentException("The planningProblem (" + planningProblem
                     + ") must not be null.");
         }
-        solverScope.setBestSolution(planningProblem);
-        outerSolvingStarted(solverScope);
-        boolean restartSolver = true;
-        while (restartSolver) {
-            solvingStarted(solverScope);
-            runPhases();
-            solvingEnded(solverScope);
-            restartSolver = checkProblemFactChanges();
+        try {
+            solverScope.setBestSolution(planningProblem);
+            outerSolvingStarted(solverScope);
+            boolean restartSolver = true;
+            while (restartSolver) {
+                solvingStarted(solverScope);
+                runPhases();
+                solvingEnded(solverScope);
+                restartSolver = checkProblemFactChanges();
+            }
+            outerSolvingEnded(solverScope);
+        } finally {
+            solving.set(false);
```

Comment:
@ge0ffrey I think that eating exception is not that bad in this case, as @dimiii explained. But I agree we need some modul to catch exceptions from planner.